app/controllers/users_controller.rb

```diff
@@ -21,7 +21,11 @@ def create
   end
 
   def show
-    @user = User.find(session[:user_id])
+    if session[:user_id]
+      @user = User.find(session[:user_id])
+    else
+      render file: "/public/404"
+    end
   end
```

Comment:
So here I think the current_user helper method actually achieves what this logic is doing, and whats really neat about the method is that you can actually just call the method in the show page html to get the users info.